src/main/java/upm/softwaredesign/finalproject/service/ActorService.java

```diff
@@ -2,83 +2,96 @@
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+
+import upm.softwaredesign.finalproject.blockchain.BlockChain;
 import upm.softwaredesign.finalproject.entity.ActorEntity;
+import upm.softwaredesign.finalproject.enums.TransactionStatus;
 import upm.softwaredesign.finalproject.model.Actor;
+import upm.softwaredesign.finalproject.model.Product;
 import upm.softwaredesign.finalproject.model.Retailer;
 import upm.softwaredesign.finalproject.repository.ActorRepository;
 
+import java.util.ArrayList;
 import java.util.List;
 
 @Service
-public class ActorService{
+public class ActorService {
 
 	private final ActorRepository repository;
-	
+
 	@Autowired
 	public ActorService(ActorRepository repository) {
 		this.repository = repository;
 	}
 
-	public Actor cast(ActorEntity actorEntity)
-	{
-		// TODO: We are loosing information about actor type here; Actor should be abstract.
-		//  TODO: We need to have a logic to distinguish type of actor and create aprioprate type
+	public Actor cast(ActorEntity actorEntity) {
+		// TODO: We are loosing information about actor type here; Actor should be
+		// abstract.
+		// TODO: We need to have a logic to distinguish type of actor and create
+		// Appropriate type
 		Actor actor = new Retailer();
 		actor.setName(actorEntity.getName());
 		actor.setType(actorEntity.getType());
 		return actor;
 	}
-	public Actor findByType(String type ) {
+
+	public Actor findByType(String type) {
```

Comment:
Is this method needed? There is now retrieveActorByType

Reply:
It was deleted